src/support_sphere/lib/logic/cubit/login_cubit.dart

```diff
@@ -13,30 +13,29 @@ class LoginCubit extends Cubit<LoginState> implements ValidatableCubit {
   final AuthenticationRepository _authenticationRepository;
 
   void emailChanged(String value) {
-    emit(
-      state.copyWith(
-        email: value,
-      ),
-    );
+    emit(state.copyWith(email: value));
   }
 
   void passwordChanged(String value) {
-    emit(
-      state.copyWith(
-        password: value,
-      ),
-    );
+    emit(state.copyWith(password: value));
   }
 
+  void toggleShowPassword() => changeShowPassword(emit, state);
+
   void setValid() => emit(state.copyWith(isValid: true));
   void setInvalid() => emit(state.copyWith(isValid: false));
 
-  void toggleShowPassword() {
-    changeShowPassword(emit, state);
+  bool isLoginButtonEnabled() {
+    return state.isValid && state.isAllFieldsFilled;
   }
```

Comment:
```suggestion
```
This looks like a computed property that is best suited to live in the state rather than a function

Reply:
Noted!